### DISCUSSION TITLE
Reexecuting useQuery not passing updated variable in Vue

### DISCUSSION THREAD

**Discussion by daviddesmet** [urql-graphql/urql#2839] [Q&A] [answered]
I've the following code using Vue 3 Composition API:

```typescript
// rest of code omitted for brevity.

const routeParamId = computed(() => route.params.id);
const result = useQuery<GetIncidentByIdQuery>({ query: GetIncidentByIdDocument, { variables: { id: routeParamId.value as string } } });

watch(
  () => route.params,
  () => {
    console.log("id", routeParamId.value);
    result.executeQuery({ requestPolicy: "network-only" });
  }
);
```

Whenever the route parameter `id` changes, the query is re-executed but is still taking into account the old variable, even when it is a computed property. The `console.log` indeed prints the updated `id`.

Are variables passed to the `useQuery` in the Composition API updated automatically? The urql's Vue documentation is not using the Composition API so I wonder if this is the correct way of doing it.

**Comment by kitten** [comment, marked as answer]
You should be able to directly derive `variables` from the input ref, i.e. give it a computed value directly, which will cause `useQuery` to correctly hook up and observe that value as it changes:

```js
const routeParamId = computed(() => route.params.id);
const result = useQuery<GetIncidentByIdQuery>({
  query: GetIncidentByIdDocument,
  variables: computed(() => ({
    id: routeParamId.value as string
  }),
});
```

Mind you, I don't 100% remember how reactive values work in Vue — we maintain multiple bindings and it's sometimes easy for me to lose track. But primarily, what's important to keep in mind is that using `value` outside of the context of something that can react to that "access" will not set up a reactive update to any changes.

So, whatever turns your variables (or other inputs) into a reactive value will be unwrapped and observed as part of an effect in `useQuery`. Unwrapping your value and mapping it inside of `computed` should achieve just this, if I'm remembering correctly, as per: https://github.com/urql-graphql/urql/blob/af93ef9ab9595478d8489a6280b08a731b1d7faa/packages/vue-urql/src/useQuery.ts#L104

**Reply by daviddesmet** [reply]
Yeah, you are right, that did the trick!
Thanks a lot :)
